### ISSUE TITLE
Only /dev/fb0 is allowed

### ISSUE THREAD

**Issue by pozzugno** [raspberrypi-ui/plymouth#1]
I'm trying to use plymouth with PiTFT small displays. They are on /dev/fb1, but plymouth-0.9.2-3+rpi1 that I installed with apt-get doesn't allow framebuffers different than /dev/fb0.

So I can't use plymouth.ignore-serial-consoles, because /dev/fb1 is not seen. However, if I don't use plymouth.ignore-serial-consoles, I have a simple text theme on /dev/tty1 (that is configured as /dev/tty0).

**Comment by spl237** [comment]
The locking to fb0 is deliberate - it is required to make the splash screen display correctly on an external monitor if a SenseHAT is being used with Pi, as otherwise the splash screen is incorrectly displayed on the SenseHAT.

I'd suggest installing the unmodified version of plymouth which is available in the standard Raspbian repo instead of plymouth-0.9.2-3+rpi1  - sudo apt-get install plymouth=0.9.2-3 should do that.

**Comment by pozzugno** [comment]
> sudo apt-get install plymouth=0.9.2-3 should do that

Unfortunately on my system:

```
pi@raspberrypi:~$ sudo apt-get install plymouth=0.9.2-3
Reading package lists... Done
Building dependency tree
Reading state information... Done
E: Version '0.9.2-3' for 'plymouth' was not found
```

**Comment by spl237** [comment]
Look at the output of apt-cache policy plymouth - it should show two version numbers, one with +rpi1 at the end and one without. Install the one without - it may have upversioned to 0.9.2-4 by now.

**Comment by spl237** [comment]
It looks like you may need to do sudo apt-get install plymouth=0.9.0-9

**Comment by pozzugno** [comment]
I installed plymouth=0.9.0-9, but it doesn't work as the newer 0.9.2-3 (that doesn't work anyway, because of /dev/fb0 issue).

```
pi@raspberrypi:~ $ sudo plymouthd --debug --kernel-command-line="splash plymouth.ignore-serial-consoles"
[main.c:1998]                        initialize_environment:initializing minimal work environment
[main.c:2017]                        initialize_environment:checking if '/dev/tty7' exists
[main.c:1847]                               check_verbosity:checking if tracing should be enabled
[main.c:1923]                               check_verbosity:tracing shouldn't be enabled!
[main.c:1943]                                 check_logging:checking if console messages should be redirected and logged
[main.c:1952]                                 check_logging:logging will be enabled!
[main.c:2031]                        initialize_environment:source built on Nov 23 2014
```
I don't know why plymouthd 0.9.0-9 checks if /dev/tty7 exists. When I start plymouthd 0.9.2-3+rpi1, the log is quite long.
```
pi@raspberrypi:~ $ sudo plymouthd --debug --kernel-command-line="splash plymouth.ignore-serial-consoles"
[main.c:1902]                        initialize_environment:initializing minimal work environment
[main.c:1916]                        initialize_environment:checking if '/dev/tty1' exists
[main.c:1764]                               check_verbosity:checking if tracing should be enabled
[main.c:1829]                               check_verbosity:tracing shouldn't be enabled!
[main.c:1848]                                 check_logging:checking if console messages should be redirected and logged
[main.c:1857]                                 check_logging:logging will be enabled!
[main.c:1929]                        initialize_environment:source built on Sep 13 2016
[main.c:1939]                        initialize_environment:Making sure /run/plymouth exists
[ply-utils.c:689]                          ply_create_directory:directory '/run/plymouth' already exists
[main.c:1943]                        initialize_environment:initialized minimal work environment
[main.c:694]                       get_cache_file_for_mode:returning cache file '/var/lib/plymouth//boot-duration'
[main.c:300]                                 load_settings:Trying to load /etc/plymouth//plymouthd.conf
[ply-key-file.c:171]                       ply_key_file_load_group:trying to load group Daemon
[ply-key-file.c:259]                      ply_key_file_load_groups:key file has no more groups
[ply-key-file.c:352]                        ply_key_file_get_value:key file does not have entry for key 'ShowDelay'
[main.c:432]                    find_system_default_splash:System configured theme file is '/usr/share/plymouth/themes/glow/glow.plymouth'
[main.c:300]                                 load_settings:Trying to load /usr/share/plymouth//plymouthd.defaults
[ply-key-file.c:171]                       ply_key_file_load_group:trying to load group Daemon
[ply-key-file.c:259]                      ply_key_file_load_groups:key file has no more groups
[main.c:320]                                 load_settings:Splash delay is set to 0.000000
[main.c:446]              find_distribution_default_splash:Distribution default theme file is '/usr/share/plymouth/themes/text/text.plymouth'
[main.c:850]           plymouth_should_show_default_splash:checking if plymouth should show default splash
[main.c:878]           plymouth_should_show_default_splash:using default splash because kernel command line has option "splash"
[ply-device-manager.c:672]                   create_seats_from_terminals:checking for consoles
[ply-device-manager.c:676]                   create_seats_from_terminals:ignoring all consoles but default console because explicitly told to.
[ply-device-manager.c:402]                         watch_for_udev_events:watching for udev graphics device add and remove events
[ply-device-manager.c:750]                 watch_for_coldplug_completion:udev coldplug completed already
[ply-device-manager.c:698]                        create_seats_from_udev:Looking for devices from udev
[ply-device-manager.c:303]                    create_seats_for_subsystem:creating seats for drm devices
[ply-device-manager.c:303]                    create_seats_for_subsystem:creating seats for frame buffer devices
[ply-device-manager.c:320]                    create_seats_for_subsystem:found device /sys/devices/platform/soc/20204000.spi/spi_master/spi0/spi0.0/graphics/fb1
[ply-device-manager.c:327]                    create_seats_for_subsystem:device is initialized
[ply-device-manager.c:336]                    create_seats_for_subsystem:found node /dev/fb1
[ply-device-manager.c:342]                    create_seats_for_subsystem:framebuffer device other than fb0 found - ignoring
[ply-device-manager.c:320]                    create_seats_for_subsystem:found device /sys/devices/platform/soc/soc:fb/graphics/fb0
[ply-device-manager.c:327]                    create_seats_for_subsystem:device is initialized
[ply-device-manager.c:336]                    create_seats_for_subsystem:found node /dev/fb0
[ply-device-manager.c:208]                   create_seat_for_udev_device:device is for local console: no
[ply-device-manager.c:220]                   create_seat_for_udev_device:device subsystem is graphics
[ply-device-manager.c:226]                   create_seat_for_udev_device:found frame buffer device /dev/fb0
[ply-device-manager.c:168]                      fb_device_has_drm_device:trying to find associated drm node for fb device (path: platform-soc:fb)
[ply-device-manager.c:191]                      fb_device_has_drm_device:no card entry!
[ply-device-manager.c:639]    create_seat_for_terminal_and_renderer_type:creating seat for /dev/fb0 (renderer type: 2) (terminal: none)
[ply-renderer.c:230]                      ply_renderer_open_plugin:trying to open renderer plugin /usr/lib/arm-linux-gnueabihf/plymouth/renderers/frame-buffer.so
[./plugin.c:257]                                create_backend:creating renderer backend for device /dev/fb0
[./plugin.c:500]                                  query_device:32 bpp (8, 8, 8, 8) with rowstride 2624
[./plugin.c:273]                               initialize_head:initializing 656x416 head
[ply-renderer.c:250]                      ply_renderer_open_plugin:opened renderer plugin /usr/lib/arm-linux-gnueabihf/plymouth/renderers/frame-buffer.so
[ply-seat.c:80]                            add_pixel_displays:Adding displays for 1 heads
[ply-seat.c:154]                                 ply_seat_open:not adding text display for seat, since seat has no associated terminal
[main.c:1011]                                 on_seat_added:listening for keystrokes
[main.c:1015]                                 on_seat_added:listening for escape
[main.c:1019]                                 on_seat_added:listening for backspace
[main.c:1023]                                 on_seat_added:listening for enter
[ply-device-manager.c:320]                    create_seats_for_subsystem:found device /sys/devices/virtual/graphics/fbcon
[ply-device-manager.c:348]                    create_seats_for_subsystem:it's not initialized
[main.c:2228]                                          main:entering event loop
pi@raspberrypi:~ $

```

And another problem is with libplymouth. I don't know if it is used by plymouth/plymouthd binaries. If yes, it is available only in 0.9.2.3+rpi1 flavor.

I see there is plymouth 0.9.2-4 available on raspbian repositories. I downloaded and tried to install through dpkg -i command, but there are some problems.
```
pi@raspberrypi:~/raspbian $ sudo dpkg -i libplymouth4_0.9.2-4_armhf.deb
(Reading database ... 39198 files and directories currently installed.)
Preparing to unpack libplymouth4_0.9.2-4_armhf.deb ...
Unpacking libplymouth4:armhf (0.9.2-4) over (0.9.2-4) ...
dpkg: dependency problems prevent configuration of libplymouth4:armhf:
 libplymouth4:armhf depends on libpng16-16 (>= 1.6.2-1); however:
  Package libpng16-16 is not installed.

dpkg: error processing package libplymouth4:armhf (--install):
 dependency problems - leaving unconfigured
Processing triggers for libc-bin (2.19-18+deb8u6) ...
Errors were encountered while processing:
 libplymouth4:armhf

```
